pyproject.toml

```diff
@@ -0,0 +1,54 @@
+[project]
+name = "soar-sml"
+description = "Raw SML Bindings to the SOAR Cognitive Architecture"
+version = "9.6.2"
+license = "BSD"
+authors = [
+    # {name = "Author Name", email = "authoremail@example.com"}
+]
+readme = "README.md"
+# Enscons does not support 3.12 as of yet
+requires-python = ">=3.8,<3.12"
+dependencies = []
+keywords = ["soar"]
+classifiers = [
+    "Programming Language :: Python :: 3",
+]
+
+[build-system]
+requires = ["pytoml>=0.1", "enscons", "wheel", "scons==4.4.0"]
+build-backend = "enscons.api"
+
+[tool.cibuildwheel]
+build-verbosity = 3
+skip = [
+    # Skip PyPy: Build works, but does not properly work at runtime.
+    "pp*",
+    # Skip 32-bit Windows: scons does not properly build 32-bit version,
+    #  and linkage problems occur.
+    "*-win32",
+    # Skip MUSL (as opposed to GLIB) linux, due to unknown concurrency symbols.
+    "*-musllinux_*"
+]
+test-command = """\
+python -c "import soar_sml;\
+k=soar_sml.Kernel.CreateKernelInNewThread();\
+a=k.CreateAgent('soar');\
+assert(a.ExecuteCommandLine('echo hello world').strip()=='hello world')"
+"""
+
+# Skip testing python 3.8 on ARM64. This is due to a limitation with cibuildwheel:
+# https://github.com/pypa/cibuildwheel/pull/1169#issuecomment-1178727618
+test-skip = "cp38-macosx_*:arm64"
+
+[tool.cibuildwheel.linux]
+repair-wheel-command = """\
+LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/project/out auditwheel repair -w {dest_dir} {wheel}\
+"""
+
+[tool.cibuildwheel.macos]
+repair-wheel-command = """\
+DYLD_LIBRARY_PATH=$DYLD_LIBRARY_PATH:$LD_LIBRARY_PATH:$REPAIR_LIBRARY_PATH:$HOME/work/Soar/Soar/out delocate-wheel \
+--require-archs {delocate_archs} -w {dest_dir} -v {wheel}\
+"""
+before-build = "rm -rf build/"
```

Comment:
scons seems to choke on a previous build's output for some reason (something about arch changes), so we just remove the build directory between builds to make it happy (and working)

Reply:
Wouldn't that mean that you have to rebuild the entire project from scratch every time? That will make the development loop significantly longer.

Reply:
This would only run in CI (with cibuildwheel), for package distribution, and only to circumvent quirks around building multiple wheels in a row.

And no, it would only rebuild the kernel every time, in addition to the python-version-specific interface file.

Reply:
Mentioning development cycles is a good point though, since I just realised that enscons does not have ([proper](https://github.com/dholth/enscons/issues/23)) support for "editable" builds (basically linking a python distribution to its original folder, where further builds can be run that'll reflect new imports immediately)

Doing that would basically have the command be `pip install -e ~/dev/soar`, with update-builds only requiring `scons soar_sml/_Python_sml_ClientInterface.so`, but I'll have to look into that.

Reply:
Done :)